packages/web-app/src/modules/error-boundary/ErrorBoundaryStore.tsx

```diff
@@ -0,0 +1,35 @@
+import { action, observable } from 'mobx'
+import type { UseErrorBoundaryApi } from 'react-error-boundary'
+
+export class ErrorBoundaryStore {
+  @observable
+  private showErrorBoundaryQueue: (() => void)[] = []
+
+  @observable
+  public errorBoundary?: UseErrorBoundaryApi<Error> = undefined
+
+  @action
+  public setErrorBoundary = (errorBoundary: UseErrorBoundaryApi<Error>) => {
+    this.errorBoundary = errorBoundary
+    // Trigger any queued showErrorBoundary actions
+    this.showErrorBoundaryQueue.forEach((queuedAction) => {
+      queuedAction()
+    })
+    this.showErrorBoundaryQueue = []
+  }
+
+  @action
+  public showErrorBoundary = (errorCaughtMessage: Error) => {
```

Comment:
What do you think about using the same naming as library does

```suggestion
  public showBoundary = (errorCaughtMessage: Error) => {
```

Reply:
I would prefer showErrorBoundary to know exactly what Boundary should be shown